lib/Sema/TypeCheckDeclOverride.cpp

```diff
@@ -937,11 +937,23 @@ static GenericSignature *getNewGenericSignature(ValueDecl *base,
   auto derivedGenericCtx = derived->getAsGenericContext();
   auto &ctx = base->getASTContext();
 
+  if (!baseGenericCtx) {
+    return nullptr;
+  }
+
   auto baseClass =
       baseGenericCtx->getInnermostTypeContext()->getSelfClassDecl();
   auto derivedClass =
       derivedGenericCtx->getInnermostTypeContext()->getSelfClassDecl();
 
+  if (!derivedClass) {
+    return nullptr;
```

Comment:
What's an example where this happens? This code should only be run for class members.

Reply:
@slavapestov I don’t have a concrete example, but these crashes were happening due to code in the standard library (collections iirc).

Reply:
You should try to reduce a test case to see what's going on. Maybe the declaration in question is a protocol member, in which case we should skip most of these checks.